pointcloud/basic_octree_utils.go

```diff
@@ -23,18 +24,26 @@ func newInternalNode(tree []*BasicOctree) basicOctreeNode {
 		children: tree,
 		nodeType: internalNode,
 		point:    PointAndData{},
+		maxProb:  -1.,
 	}
 	return octNode
 }
 
 // Creates a new LeafNodeFilled and stores specified position and data.
-func newLeafNodeFilled(p r3.Vector, d Data) basicOctreeNode {
+func newLeafNodeFilled(p r3.Vector, d Data) (basicOctreeNode, error) {
+	// if !d.HasValue() {
```

Comment:
leaving this commented out for now.
Having it in action causes for `testPCDToBasicOctree` to fail.

Reply:
@jeremyrhyde , thoughts on whether this error is better/more efficienct to throw here or in `getMaxProb`?

Reply:
Additionally, consider if it's worth breaking out these checks into a helper function that takes in `pointcloud.Data` and outputs a float.